sakura_core/util/design_template.h

```diff
@@ -62,7 +62,7 @@ class TSingleton{
 	}
 
 protected:
-	TSingleton(){}
+	TSingleton() = default;
```

Comment:
I'd like to deprecate this class.

See [boost-di](https://boost-ext.github.io/di/#stupid-vs-solid-clean-code-uncle-bob).

Reply:
@berryzplus revert here https://github.com/sakura-editor/sakura/pull/1860/commits/1fc1801a340c0d1ebb2b306d27b3d48c14ec2c17